Agent designer tutorial first draft (#14)
silverpond/highlighter-docs: content/docs/tutorials/creating_and_editing_agents.md

+++
title = "Creating and Editing Your Agents"
description = "Creating and Editing Your Agents"
date = 2021-05-01T08:00:00+00:00
updated = 2021-05-01T08:00:00+00:00
draft = false
weight = 100
sort_by = "weight"
template = "docs/page.html"

[extra]
lead = 'Learn to use the Highlighter Agent Designer'
toc = true
top = false
+++

### Why

> Highlighter enables you identify high-level entities and attributes in low-level data. This work is delegated to "agents", which could be a human annotating images, video or text, or it could be a set of machine processes, called "capabilities". Highlighter's Agent Designer is where you assemble your capabilities into an Agent that can process your data.

Reviewer: I guess "agents' should be capitalised to match other uses

Author: Done

In this tutorial, you'll create a new Agent, add Capabilities, and connect them all up to make your Agent ready to deploy.

### Create an Agent

Before we create a new Agent, we need an Assessment Process to add it to. The following steps will lead you through creating a new Assessment Process and a new Agent:
1. Navigate to [highlighter.ai/projects](https://highlighter.ai/projects) and click "New Assessment Process" (we'll delete the new Process at the end).

> 2. Give your new Process a name, ignore the rest of the form and click "Save Assessment Process".

Reviewer: could delete this
```suggestion
2. Give your new Process a name and click "Save Assessment Process".
```

Author: Done

> 3. Navigate to the "Stage Definition" tab of your new Process and under "Machine Assessments" press "New Machine Assessment Stage".

Reviewer: ```suggestion
3. Navigate to the "Stage Definition" tab of your new Process and under "Machine Assessments" click "New Machine Assessment Stage".
```

Author: done

4. Give your new Stage a name and click "Save Machine Assessment Stage".
5. Looking at your new Machine Assessment Stage, click "Edit" on the top right of the page. You're now in the Agent Designer!

### Add Capabilities

> You should now be looking at a blank agent. By default it has two nodes: Inputs and Outputs. You can drag these around.

Reviewer: Your screenshot has "Pipeline Outputs" so probably should use that, unless you've changed that to new terminology in which case screenshot needs updating

Author: Good point. I'll have to remember to update this when I migrate "Pipeline Inputs" to "Inputs".

{{ resize_image(path="docs/tutorials/new-agent.png", width=700, height=1, op="fit_width") }}

> To add a Capability to the Agent, click on "Add Capability" in the top right. You will see a list of Capabilities; select a "detector" Capability from the list. Clicking the detector will add it to the Agent as a node that you can drag around.

Reviewer: Needs space between image and this paragraph

Author: Done

To add edges between the nodes, hover your mouse over the Inputs node. An output port will appear on the right, which you can grab and drag towards the input of the detector. Dropping the new edge on the input port of the detector will add the edge to the Agent.

> You can add edges to the Outputs node by dragging the outputs of the detector and dropping the new edge on the Outputs node. These edges represent data returned by your Agent.

Reviewer: ?
```suggestion
You can add edges to the Outputs node by dragging the outputs of the detector and dropping the new edge on the Outputs node. These edges represent data that your Agent will return.
```

The new ports on the Inputs and Outputs nodes are given the names of the ports you have connected them to, but you can edit these names by clicking on them.

{{ resize_image(path="docs/tutorials/new-agent-with-detector.png", width=700, height=1, op="fit_width") }}

### Configure your Capabilities

Double-click on the detector Capability, or click the hamburger icon next to its title. This will bring up the detailed attributes and editable parameters of the Capability.

You can remove a Capability from your agent by clicking "Remove Capability" on this detailed view.

> ### Tidy Up

Reviewer: Not sure we even need this section? I assuming someone will want to keep their agent

Author: Since this is a "tutorial" rather than a "how-to guide", the idea is someone runs through it to learn "how" to add an agent, rather than running through it when they just want an agent created. https://diataxis.fr/tutorials-how-to/

This might be more useful as a "how-to guide", in which case I would remove the setup section as well as the user should already have an Assessment Process by the time they reach for a guide to help them create an agent.

What do you reckon?

Author: I've added a note about ignoring the section if the user wants to use their new Assessment Process and Agent

Reviewer: Yeh I reckon this is actually more of a how-to guide than a tutorial anyway. There is a 'how-to- guides section on main now so you could just add it to that

Author: Ok I've made it a How-To Guide

To add edges between the nodes, hover your mouse over the "Pipeline Inputs" node. An output port will appear on the right, which you can grab and drag towards the input of the detector. Dropping the new edge on the input port of the detector will add the edge to the Agent.

You can add edges to the "Pipeline Outputs" node by dragging the outputs of the detector and dropping the new edge on the "Pipeline Outputs" node. These edges represent data that your Agent will return.

The new ports on the "Pipeline Inputs" and "Pipeline Outputs" nodes are given the names of the ports you have connected them to, but you can edit these names by clicking on them.

{{ resize_image(path="docs/tutorials/new-agent-with-detector.png", width=700, height=1, op="fit_width") }}

### Configure your Capabilities

Double-click on the detector Capability, or click the hamburger icon next to its title. This will bring up the detailed attributes and editable parameters of the Capability.

You can remove a Capability from your Agent by clicking "Remove Capability" on this detailed view.

### Tidy Up

> Ignore this section if you want to continue using your new Assessment Process an Agent.

Reviewer: ```suggestion
Ignore this section if you want to continue using your new Assessment Process and Agent.
```
